official/vision/beta/modeling/layers/nn_layers.py

```diff
@@ -730,7 +731,11 @@ def _compute_buffered_causal_padding(self,
          (self.kernel_size[i] - 1) * (self.dilation_rate[i] - 1))
         for i in range(self.rank)
     ]
-    pad_total = [kernel_size_effective[i] - 1 for i in range(self.rank)]
+    pad_total = [max(kernel_size_effective[i] - (self.strides[i]), 0)
```

Comment:
This should only be applied after the first temporal dimension, otherwise the causal padding won't match with `tf.nn.conv1d(padding='causal')`.

```suggestion
    pad_total = [kernel_size_effective[0] - 1]
    for i in range(1, self.rank):
      overlap = (input_shape[i] - 1) % self.strides[i] + 1
      pad_total.append(max(kernel_size_effective[i] - overlap, 0))
```

Reply:
couldn't this cause to not attend to some of the input also for the standard implementation?